pkg/uidmap/uidmap.go

```diff
@@ -0,0 +1,136 @@
+package uidmap
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/docker/libcontainer/user"
+)
+
+func xlateOneFile(path string, finfo os.FileInfo, containerRoot uint32, inverse bool) error {
+	uid := uint32(finfo.Sys().(*syscall.Stat_t).Uid)
+	gid := uint32(finfo.Sys().(*syscall.Stat_t).Gid)
+	mode := finfo.Mode()
+
+	if (uid == containerRoot || gid == containerRoot) && !inverse {
+		fmt.Errorf("Warning: UID of an existing file (%s) is docker-root")
+	}
+	if ((uid == 0 || gid == 0) && !inverse) || ((uid == containerRoot || gid == containerRoot) && inverse) {
```

Comment:
If that's a rarely used function, My recommendation is to separate that into an "xLateOnefileInReverse" method, for simplicity.